### PR TITLE
Add Java 17 to build/test matrix

### PR DESCRIPTION
**Pull request by basil**

### CODE REVIEW

**Comment by basil** [comment]
Used my admin powers to replay the build with the `Jenkinsfile` changes here: https://ci.jenkins.io/job/Plugins/job/saml-plugin/job/PR-340/2/

File: Jenkinsfile [hunk 1]
```diff
@@ -1,7 +1,6 @@
 #!/usr/bin/env groovy
 
-buildPlugin(useContainerAgent: true,
-        configurations: [
-                [platform: 'linux', jdk: '11'],
-                [platform: 'windows', jdk: '11']
-        ])
+buildPlugin(useContainerAgent: true, configurations: [
+  [platform: 'linux', jdk: 17],
```
Comment (kuisathaverat) [inline Jenkinsfile:4]:
```suggestion
  [platform: 'linux', jdk: 17],
  [platform: 'linux', jdk: 11],
```

Comment (kuisathaverat) [reply Jenkinsfile:4]: I'd preferred to not remove Linux Java 11 build and test

Comment (basil) [reply Jenkinsfile:4]: Inconsistent with the archetype, but OK

**Review by kuisathaverat — APPROVED** [review]

**Pull request merged**